lib/tree.rb

```diff
@@ -16,51 +32,84 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # add(value) - adds a value to the Binary Search Tree
+  # Time Complexity: O(n)
+  # Space Complexity: O(1)
   def add(key, value)
```

Comment:
👍 , but since you are doing recursion the time/space complexity are **both** O(n) if the tree is unbalanced and O(log n) if it is balanced.